xarray/tests/test_backends.py

```diff
@@ -1647,17 +1654,82 @@ def test_encoding_kwarg_fixed_width_string(self):
     def test_dataset_caching(self):
         super(CFEncodedBase, self).test_dataset_caching()
 
-    @pytest.mark.xfail(reason="Zarr stores can not be appended to")
     def test_append_write(self):
-        super(CFEncodedBase, self).test_append_write()
+        ds, ds_to_append, _ = create_append_test_data()
+        with self.create_zarr_target() as store_target:
+            ds.to_zarr(store_target, mode='w')
+            ds_to_append.to_zarr(store_target, mode='a', append_dim='time')
+            original = xr.concat([ds, ds_to_append], dim='time')
+            assert_identical(original, xr.open_zarr(store_target))
 
     @pytest.mark.xfail(reason="Zarr stores can not be appended to")
     def test_append_overwrite_values(self):
```

Comment:
what is this test exactly supposed to do? Is it something like what's happening here: https://github.com/pydata/xarray/blob/master/xarray/tests/test_backends.py#L881?
If so,this kind of functionality is not implemented in this PR yet. You can only "append" to existing variables, not overwrite existing data. That would have to be done with the 'w' mode.

Reply:
Yes, I think you have  analyzed this correctly. This  is fine.